utils/esr_peer_info.cpp

```diff
@@ -0,0 +1,382 @@
+        // will be correctly null terminated for the boost::algorithm::unhex function
+        char line[ESR_RESPONSE_LINE_LENGTH + 1]{0};
+        std::istringstream stream{result.begin()};
+        while(stream.read(line, ESR_RESPONSE_LINE_LENGTH))
+        {
+            switch (state)
+            {
+                case NODE_COUNT: state = NA_0; break;
+                case NA_0: state = HTTP_PORT; break;
+                case HTTP_PORT:
+                    http_port = uint16_t(std::strtoul(line, nullptr, 16));
+                    if (0 == http_port)
+                    {
+                        LOG(warning) << "Invalid value for http port:[" << http_port << "] node may not exist";
+                    }
+                    state = NA_1;
+                    break;
+                case NA_1: state = NODE_PORT; break;
+                case NODE_PORT:
+                    port = std::strtoul(line, nullptr, 16);
+                    if (0 == http_port)
+                    {
+                        LOG(warning) << "Invalid value for port:[" << port << "] node may not exist";
+                    }
+                    state = NA_2;
+                    break;
+                case NA_2:
+                    state = NODE_HOST;
+                    break;
+                case NODE_HOST:
+                    host = hex_to_char_string(line);
+                    trim_right_nulls(host);
+                    if (0 == http_port)
+                    {
+                        LOG(warning) << "Empty value for host ip, node may not exist";
+                    }
+                    state = NA_3;
+                    break;
+                case NA_3:
+                    state = NODE_NAME;
+                    break;
+                case NODE_NAME:
+                    name = hex_to_char_string(line);
+                    trim_right_nulls(name);
+                    if (0 == http_port)
+                    {
+                        LOG(warning) << "Empty value for host name, node may not exist";
+                    }
+                    break;
+                default:
+                    LOG(error) << "Failed to correctly parse peer info from esr";
+                    return bzn::peer_address_t(host, port, http_port, name, peer_id);
+                    break;
+            }
+        }
+        return bzn::peer_address_t(host, port, http_port, name, peer_id);
+    }
+
+
+    std::string
+    pad_str_to_mod_64(const std::string &parameter)
+    {
+        const size_t REQUIRED_MOD{64};
+        const size_t PADDING_REQUIRED = (REQUIRED_MOD - parameter.size() % REQUIRED_MOD);
+        std::string result{parameter};
+        result.insert(result.size(), PADDING_REQUIRED, '0');
+        return result;
+    }
+
+
+    std::string
+    size_type_to_hex(size_t i, size_t width = 8)
+    {
+        std::stringbuf buf;
+        std::ostream os(&buf);
+        os << std::setfill('0') << std::setw(width)
```

Comment:
why wrap this line?

Reply:
I was trying to make it look nice. But I've changed it to one line.